core/src/main/java/org/elasticsearch/index/similarity/ScriptedSimilarity.java

```diff
@@ -0,0 +1,284 @@
+                            Explanation.match(weight.field.sumDocFreq, "field.sumDocFreq"),
+                            Explanation.match(weight.field.sumTotalTermFreq, "field.sumTotalTermFreq"),
+                            Explanation.match(term.docFreq, "term.docFreq"),
+                            Explanation.match(term.totalTermFreq, "term.totalTermFreq"),
+                            Explanation.match(freq.getValue(), "doc.freq", freq.getDetails()),
+                            Explanation.match(doc.getLength(), "doc.length"));
+                }
+            };
+        }
+        if (scorers.length == 1) {
+            return scorers[0];
+        } else {
+            // Sum scores across terms like a BooleanQuery would do
+            return new SimScorer() {
+
+                @Override
+                public float score(int doc, float freq) throws IOException {
+                    double sum = 0;
+                    for (SimScorer scorer : scorers) {
+                        sum += scorer.score(doc, freq);
+                    }
+                    return (float) sum;
+                }
+
+                @Override
+                public float computeSlopFactor(int distance) {
+                    return 1.0f / (distance + 1);
+                }
+
+                @Override
+                public float computePayloadFactor(int doc, int start, int end, BytesRef payload) {
+                    return 1f;
+                }
+
+                @Override
+                public Explanation explain(int doc, Explanation freq) throws IOException {
+                    Explanation[] subs = new Explanation[scorers.length];
+                    for (int i = 0; i < subs.length; ++i) {
+                        subs[i] = scorers[i].explain(doc, freq);
+                    }
+                    return Explanation.match(score(doc, freq.getValue()), "Sum of:", subs);
+                }
+            };
+        }
+    }
+
+    private static class Weight extends SimWeight {
+        final String fieldName;
+        final Query query;
+        final Field field;
+        final Term[] terms;
+
+        Weight(String fieldName, Query query, Field field, Term[] terms) {
+            this.fieldName = fieldName;
+            this.query = query;
+            this.field = field;
+            this.terms = terms;
+        }
+    }
+
+    /** Scoring factors that come from the query. */
+    public static class Query {
+        final float boost;
+
+        private Query(float boost) {
+            this.boost = boost;
+        }
+
+        /** The boost of the query. It should typically be incorporated into the score as a multiplicative factor. */
+        public float getBoost() {
+            return boost;
+        }
+    }
+
+    /** Statistics that are specific to a given field. */
+    public static class Field {
+        final long docCount;
+        final long sumDocFreq;
```

Comment:
These could be private since the getters are public? Or the getters could be removed and just make these public final?